### DISCUSSION TITLE
How to run gluetun through powershell?

### DISCUSSION THREAD

**Discussion by testinghere57564** [qdm12/gluetun#817] [Q&A]
Hello, as I understand that after installing Docker Desktop, I have the opportunity to run containers through powershell. I'm writing
```
docker run -it --rm --cap-add=NET_ADMIN -e VPNSP=nordvpn \
-e OPENVPN_USER=validusername (OPENVPN_USER=marcel.menne@gmx.de) -e OPENVPN_PASSWORD=validpass \
-e HTTPPROXY=on \
-p 8122:8888 \
-e REGION=Netherlands qmcgaw/gluetun
```

 but I'm getting the error **docker: invalid reference format**. Apparently powershell requires some other syntax. Please tell me how to fix this?

**Comment by testinghere57564** [comment]
I want to make a script that would launch different VPNs, since it takes a long time to manually enter data each time.

**Comment by qdm12** [comment]
You probably need to surround your OPENVPN_USER value and OPENVPN_PASSWORD value with double quotes. In example

```
-e OPENVPN_USER="marcel.menne@gmx.de" -e OPENVPN_PASSWORD="validpass"
```

I am also using powershell core and it works fine over here.

**Reply by testinghere57564** [reply]
Unfortunately, it didn't help. I can only use gluetun in Docker Desktop, but I can't run this container in powershell. Powershell says it doesn't understand the -it , --rm flags. Perhaps you need to download some kind of add-on? I can run gluetun without any settings in powershell, but it crashes on a greeting because of this. Some standard containers from microsoft run in powershell without errors. I'm new to docker, I'm trying to figure it out, because I really need a similar solution, but I can't solve the problem, I couldn't find anything on the Internet about this.

**Reply by qdm12** [reply]
Can you run `docker run -it --rm alpine` for example? What terminal are you using by the way? I use the Microsoft windows terminal with powershell 7.1.3

**Reply by testinghere57564** [reply]
From C:\WINDOWS\system32> host
Powershell Version  : 5.1.19041.1320

![image](https://user-images.githubusercontent.com/91505572/151512842-fb18739f-136e-426e-991c-90b121cbb0ce.png)
wrong about -it , --rm flags. I do not know what to do.
![image](https://user-images.githubusercontent.com/91505572/151513353-c26afb7c-2e2e-4627-833c-f0b38fb04561.png)

**Reply by testinghere57564** [reply]
\ were redundant for launching via powershell. Same as enter.Managed to launch! Thanks for the great work!
